src/software/ai/hl/stp/action/action_world_params_update_visitor.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "software/ai/hl/stp/action/action_visitor.h"
+#include "software/ai/hl/stp/action/all_actions.h"
+#include "software/world/world.h"
+
+/**
+ * This class can be used to generically update the world parameters of any action,
+ * without knowing the type of the action
+ */
+class ActionWorldParamsUpdateVisitor : public ActionVisitor
```

Comment:
uber nit: did we want to put `Action` in the class name? Since its already extending `ActionVisitor`
maybe `class WorldParamsUpdateVisitor : public ActionVisitor` would suffice?

Reply:
Thought about that, but we have world params for other things like tactics, so I think this is required to avoid ambiguity.

Reply:
In fact, `TacticWorldParamsUpdateVisitor` already exists, and we should be using it.......... one sec.

Reply:
OK! So I renamed `ActionVisitor` to `MutableActionVisitor`, but also realized I totally forgot one of the main wins of this issue, used the tactic update visitor in the main STP loop to remove all tactic world updates in plays!